### DISCUSSION TITLE
How to calculate product volumetric weight instead the product weight?

### DISCUSSION THREAD

**Discussion by abdo0** [saleor/saleor#14826] [Q&A] [answered]
Hi

We have a Saleor React on production (dashboard => v3.3, core => v3.1), we want to calculate the volumetric weight for each product on the cart and summarize all as the shipment method will charge us for volumetric weight instead of weight only. is there any way to for it?

**Comment by timuric** [comment, marked as answer]
Hey @abdo0 you can create product attributes for storing volumetric data. To calculate shipping costs you can use this webhook https://docs.saleor.io/docs/3.x/developer/extending/webhooks/synchronous-events/shipping

**Reply by abdo0** [reply]
Hey @timuric it only provides a weight calculation, not a width-height-depth-weight calculation

**Reply by timuric** [reply]
@abdo0 you need to extend the product types with number attribute (See configuration Product types) which allows you to extend the product model

**Reply by abdo0** [reply]
going to give that a try
